llvmlite 0.46 breaks optimization

When running the artiq compiler with llvmlite==0.46 (instead of the 0.45 used in the current nix environment), the python process crashes with the exception below.
To reproduce run an arbitrary experiment with `artiq_run`.  

```
Invalid optimization level for size level 2
UNREACHABLE executed at /tmp/pip-install-34zfw1k3/llvmlite_3325acc1594f407ebffc2ad8ceb9ded2/ffi/newpassmanagers.cpp:286!
Fatal Python error: Aborted

Current thread 0x00007f891f0f5b28 (most recent call first):
  File "/usr/lib/python3.12/site-packages/llvmlite/binding/ffi.py", line 212 in __call__
  File "/usr/lib/python3.12/site-packages/llvmlite/binding/newpassmanagers.py", line 585 in getModulePassManager
  File "/git/artiq/artiq/compiler/targets.py", line 120 in optimize
  File "/git/artiq/artiq/compiler/targets.py", line 151 in compile
  File "/git/artiq/artiq/compiler/targets.py", line 188 in compile_and_link
  File "/git/artiq/artiq/coredevice/core.py", line 150 in compile
  File "/git/artiq/artiq/coredevice/core.py", line 175 in run
  File "/git/artiq/artiq/language/core.py", line 54 in run_on_core
```

If i change the size optimization level [here](https://github.com/m-labs/artiq/blob/c2f17af93abae2b72cc6b5a12d9fcd2d784cdfc7/artiq/compiler/targets.py#L111) to `size_level=0` the code compiles without issue. The same goes for when changing the [speed level ](https://github.com/m-labs/artiq/blob/c2f17af93abae2b72cc6b5a12d9fcd2d784cdfc7/artiq/compiler/targets.py#L117) to `2` (or commenting out the line). 

If one sets the speed level directly in line 111:
```
pto = llvm.create_pipeline_tuning_options(speed_level=1, size_level=1)
```
It directly complains:
```
self = <llvmlite.binding.newpassmanagers.PipelineTuningOptions object at 0x7f3373ff8d70>, value = 1

    @size_level.setter
    def size_level(self, value):
        if not 0 <= value <= 2:
            raise ValueError("Optimization level for size should be 0, 1, or 2")
        if value != 0 and self.speed_level != 2:
>           raise ValueError(
                "Optimization for size should be encoded with speed level == 2")
E           ValueError: Optimization for size should be encoded with speed level == 2

/usr/lib/python3.12/site-packages/llvmlite/binding/newpassmanagers.py:519: ValueError
```

Seems like transitioning to the NewPassManager is still ongoing in llvmlite...

--- a/artiq/compiler/targets.py
+++ b/artiq/compiler/targets.py
@@ -1,6 +1,6 @@
 import os, sys, tempfile, subprocess, io
 from artiq.compiler import types, ir
-from llvmlite import ir as ll, binding as llvm
+from llvmlite import ir as ll, binding as llvm, __version__ as llvmlite_version
 
 llvm.initialize_all_targets()
 llvm.initialize_all_asmprinters()
@@ -108,13 +108,18 @@ class Target:
 
     def optimize(self, llmodule):
         llmachine = self.target_machine()
-        pto = llvm.create_pipeline_tuning_options(size_level=1)
+        if tuple(int(s) for s in llvmlite_version.split(".")) > (0, 45, 0):
+            pto = llvm.create_pipeline_tuning_options(speed_level = 2, size_level=1)
+        else:
+            pto = llvm.create_pipeline_tuning_options(size_level=1)
 
-        # This combination of levels gets mapped to the -Os PassManagerBuilder default in
-        # llvmlite 0.45. The mapping is slightly nonsensical in that -Oz (which is more
-        # aggressively optimizing for size at the expense of runtime) is accessed via
-        # speed_level=size_level=2.
-        pto.speed_level = 1
+            # This combination of levels gets mapped to the -Os PassManagerBuilder default in
+            # llvmlite 0.45. The mapping is slightly nonsensical in that -Oz (which is more
+            # aggressively optimizing for size at the expense of runtime) is accessed via
+            # speed_level=size_level=2.
+            # This was fixed in an incompatible way in llvmlite 0.46 breaking this workaround
+            # here, requiring a second workaround to check the llvmlite version...
+            pto.speed_level = 1
 
         pb = llvm.create_pass_builder(llmachine, pto)
         llpassmgr = pb.getModulePassManager()

